src/ray/object_manager/connection_pool.cc

```diff
@@ -85,6 +96,21 @@ void ConnectionPool::Remove(ReceiverMapType &conn_map, const ClientID &client_id
   connections.erase(connections.begin() + pos);
 }
 
+void ConnectionPool::Remove(SenderMapType &conn_map, const ClientID &client_id,
+                            const std::shared_ptr<SenderConnection> &conn) {
+  auto it = conn_map.find(client_id);
+  if (it == conn_map.end()) {
+    return;
+  }
+  auto &connections = it->second;
+  int64_t pos =
```

Comment:
Should we have a check that conn actually is in the vector?

I’m a bit surprised by the arithmetic. There isn’t a simpler way to do it?

Reply:
I have ideas for how to refactor this whole class after 0.6, but for now, just copied the current code.